README.md

```diff
@@ -2,83 +2,119 @@
 
+## <a name="implementing-ecs"></a>Implementing ECS
 
-| Field  | Description  | Type  | Multi Field  | Example  |
-|---|---|---|---|---|
-| <a name="@timestamp"></a>`@timestamp`  | Timestamp when the event was created.<br/>For log events this is expected to be when the event was generated and not when it was read.<br/>Timestamp is a required field and must exist in all events.  | date  |   | `2016-05-23T08:05:34.853Z`  |
-| <a name="tags"></a>`tags`  | Tags is a list of keywords which are used to tag each event.  | keyword  |   | `["production", "env2"]`  |
-| <a name="labels"></a>`labels`  | Labels is an object which contains key/value pairs.<br/>Labels can be used to add additional meta information to events. Label should not contain nested objects and all values are stored as keyword.<br/>An example usage is the docker and k8s labels.  | object  |   | `{key1: value1, key2: value2}`  |
-| <a name="message"></a>`message`  | For log events the message field contains the log message.<br/>In other use cases the message field can be used to concatenate together different values which are then freely searchable. Or if multiple messages exist they can be combined here into one message.  | text  |   | `Hello World`  |
+**Guidelines**
+
+* The document MUST have the `@timestamp` field.
+* The [data type](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/mapping-types.html) defined for an ECS field MUST be used.
+* It SHOULD have the field `event.version` to define which version of ECS it uses.
+* As many fields as possible should be mapped to ECS.
 
+**Writing fields**
 
-## <a name="agent"></a> Agent fields
+* All fields must be lower case
+* No special characters except `_`
+* Use underscore to combine words
 
-The agent fields contains the data about the agent/client/shipper that created the event.
+**Naming Fields**
 
-As an example in case of Beats for logs the `agent.name` is `filebeat`. In the case of APM it is the agent running in the app / service. The agent information does not change if data is sent through queuing system like Kafka, Redis, or processing systems like Logstash or APM Server.
+* *Present tense.* Use present tense unless field describes historical information.
+* *Singular or plural.* Use singular and plural names properly to reflect the field content. For example, use `requests_per_sec` rather than `request_per_sec`.
+* *General to specific.* Organise the prefixes from general to specific to allow grouping fields into objects with a prefix like `host.*`.
+* *Avoid repetition.* Avoid stuttering of words. If part of the field name is already in the prefix, do not repeat it. Example: `host.host_ip` should be `host.ip`.
+* *Use prefixes.* Fields must be prefixed except for the base fields. For example all `host` fields are prefixed with `host.`. See `dot` notation in FAQ for more details.
+* Do not use abbreviations (few exceptions like `ip` exist)
+
+## <a name="fields"></a>Fields
+
+[Base fields](#base) 
```

Comment:
I need to adjust the script to create this list.

Reply:
@ruflin  Is this helpful enough to warrant the effort?  The intention is  a compact, clickable list to reduce need for scrolling.